### PR TITLE
 Bug fix : Coarse patch dx for external field init

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
The development branch with mesh-refinement was showing incorrect values for the field with a single-level of refinement

This PR fixes the dx for the coarse-patch used to compute the coordinates that are passed to the parser.

Here is the input file 
[inputs_3d_Bfield.txt](https://github.com/ECP-WarpX/WarpX/files/10582191/inputs_3d_Bfield.txt)

Development branch : 
Screenshot from 2023-02-03 10-59-33

Fix : (The value of the field must be equal to x, where x = [0,1] )
Screenshot from 2023-02-03 10-59-45

Thanks to Lizzette Corrales for helping uncover this!

### CODE REVIEW

**Comment by lzcorrales** [comment]
Development Branch: (Bx did not equal x on level 1)
3d_before_fixed_bug

Fix: (the value of Bx=x  on both levels)
3d_after_fixed_bug

File: Source/Initialization/WarpXInitData.cpp [hunk 1]
```diff
@@ -876,10 +876,16 @@ WarpX::InitializeExternalFieldsOnGridUsingParser (
        std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& edge_lengths,
        std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& face_areas,
        const char field,
-       const int lev)
+       const int lev, PatchType patch_type)
 {
 
-    const auto dx_lev = geom[lev].CellSizeArray();
+    auto dx_lev = geom[lev].CellSizeArray();
+    amrex::IntVect refratio = ( (lev > 0 ) ? WarpX::RefRatio(lev-1) : amrex::IntVect(1) );
```
Comment (ax3l) [inline Source/Initialization/WarpXInitData.cpp]: Operator precedence in C++ makes the `()` superfluous :)
```suggestion
    amrex::IntVect refratio = (lev > 0 ) ? WarpX::RefRatio(lev-1) : amrex::IntVect(1);
```
https://en.cppreference.com/w/cpp/language/operator_precedence

**Comment by ax3l** [comment]
@RevathiJambunathan does the `PEC_field_mr` test need a reset of the checksum now? :)

https://github.com/ECP-WarpX/WarpX/blob/9411576a014d477ec24aa957a0402d16ec9aa5f8/Regression/WarpX-tests.ini#L3203-L3218

https://github.com/ECP-WarpX/WarpX/blob/9411576a014d477ec24aa957a0402d16ec9aa5f8/Examples/Tests/pec/inputs_field_PEC_mr_3d#L40-L48

**Comment by RevathiJambunathan** [comment]
Hi @ax3l 
Yes. 
That test uses external field initialization with MR.
I will reset that.

**Review by ax3l — APPROVED** [review]
Thank you Reva, Lizzette, Hannah and all involved!

**Pull request merged**